media-libs/loadpng/loadpng-1.5.recipe

```diff
@@ -1,33 +1,48 @@
-DESCRIPTION="
-loadpng is some glue that makes it easy to use libpng to load and save bitmaps \
-from Allegro programs.
-"
+SUMMARY="Load and save bitmaps from Allegro programs"
+DESCRIPTION="loadpng is some glue that makes it easy to use libpng to load \
+and save bitmaps from Allegro programs."
 HOMEPAGE="http://tjaden.strangesoft.net/loadpng/"
-SOURCE_URI="http://tjaden.strangesoft.net/loadpng/loadpng-1.5.tar.gz"
+COPYRIGHT="1997-2004 Psyk Software."
+LICENSE="Public Domain"
 REVISION="1"
-STATUS_HAIKU="stable"
-DEPEND="media-libs/allegro >= 4.4
-		media-libs/libpng >=1.2"
-MESSAGE="This might not build with libpng >= 1.4"
-CHECKSUM_MD5="5e9ecdf1762c3fca47477bf0263b96ba"
+SOURCE_URI="https://tjaden.strangesoft.net/loadpng/loadpng-1.5.tar.gz"
+CHECKSUM_SHA256="ba0e496e5368de70e6df43cb5500468cda402e63ab95b6a466e29ec6b795b7d7"
+PATCHES="loadpng-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86_64"
+
+PROVIDES="
+	loadpng = $portVersion
+	devel:libldpng = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:liballeg
+	lib:libpng16
+	lib:libz
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:liballeg
+	devel:libpng16
+	devel:libz
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
 BUILD()
 {
-	cd loadpng-1.5
-	make
+	make $jobArgs
```

Comment:
It's also possible to build a shared lib, but I don't know if it's a good idea.

    make $jobArgs shared

Reply:
There is an option to build a shared file, but even the author didn't wan't to use that as a default one?